src/main/java/de/tum/in/www1/artemis/config/Constants.java

```diff
@@ -37,25 +37,13 @@ public final class Constants {
 
     public static final String NEW_RESULT_TOPIC = "/topic/newResults";
 
-    public static final String NEW_RESULT_RESOURCE_PATH = "programming-exercises/new-result";
+    public static final String NEW_RESULT_RESOURCE_API_PATH = "/api/public/programming-exercises/new-result";
 
-    public static final String NEW_RESULT_RESOURCE_API_PATH = "/api/" + NEW_RESULT_RESOURCE_PATH;
+    public static final String TEST_CASE_CHANGED_API_PATH = "/api/programming-exercises/test-cases-changed/";
 
-    public static final String TEST_CASE_CHANGED_PATH = "/programming-exercises/test-cases-changed/";
+    public static final String PROGRAMMING_SUBMISSION_RESOURCE_API_PATH = "/api/programming-submissions/";
 
-    public static final String TEST_CASE_CHANGED_API_PATH = "/api" + TEST_CASE_CHANGED_PATH;
-
-    public static final String PROGRAMMING_SUBMISSION_RESOURCE_PATH = "/programming-submissions/";
-
-    public static final String PROGRAMMING_SUBMISSION_RESOURCE_API_PATH = "/api" + PROGRAMMING_SUBMISSION_RESOURCE_PATH;
-
-    public static final String ATHENE_RESULT_PATH = "/athene-result/";
-
-    public static final String ATHENE_RESULT_API_PATH = "/api" + ATHENE_RESULT_PATH;
-
-    public static final String SYSTEM_NOTIFICATIONS_RESOURCE_PATH = "/system-notifications/";
-
-    public static final String SYSTEM_NOTIFICATIONS_RESOURCE_PATH_ACTIVE_API_PATH = "/api" + SYSTEM_NOTIFICATIONS_RESOURCE_PATH + "active";
+    public static final String ATHENE_RESULT_API_PATH = "/api/athene-result/";
```

Comment:
Should those get the `/public/` part as well, or are those the definitions of the deprecated endpoints?
⇒ either update the URL or add a comment about that. (can constants be `@Deprecated`?)

Edit: The `LegacyResource` does not seem to use them. Are they used?

Reply:
Constants can be deprecated, yes. However, I just replaced the one occurrence of the constant with the value and removed it then.